examples/tfengine/generated/devops/devops/variables.tf

```diff
@@ -0,0 +1,59 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+variable "billing_account" {
+  type        = string
+  description = "ID of billing account to attach to this project."
+}
+
+variable "parent_id" {
+  type        = string
+  description = "ID of parent GCP resource to apply the policy. Can be one of the organization ID or folder ID according to parent_type."
+  validation {
+    condition     = can(regex("^[0-9]{8,25}$", var.parent_id))
+    error_message = "The parent_id must be valid. Should have only numeric values with a length between 8 and 25 digits."
+  }
+}
+
+variable "project" {
+  type = object({
+    apis = list(string)
+    owners_group = object({
+      customer_id  = string
```

Comment:
>Just looking at `admins_group`, I don't see why the description should be different. The extra sentence is useful. Maybe that's a bug?

Maybe bad example. I see it useful too, but I just wanted to show that could be particular cases in which the description may be different. Nothing so bad at the moment

>To answer your questions:
>>Should we add descriptions to sub-variables?
>
>Yes, I think it's useful. It's unfortunate that they can't have a "description" field but I suggest: 1) Add a comment above each one 2) If not too difficult, add them to the main variable's description, on a separate line, since this way they show up when terraform prints information about the variable (comments are ignored so a person would have to open the variables.tf file). If the latter is difficult to do right, then stick to just comments.

Personally, I like more approach #1. Seems more consistent for me, and otherwise, we should define a standard to convert the naming for nested values to main values. I already said that in another comment, but I find it too difficult to maintain.

Maybe the resolution of this will depend on how important is it to make the terraform outputs to show the description?

>>Should the descriptions be the same as in the schema?
>
>My opinion is yes.
>
>>Isn't it difficult to maintain if they're in separate places?
>
>Yes, it is. I don't see a good way around it right now, since not all the options in the recipe schema correspond to terraform vars.
>
>>Many schema variables don't end up written here.
>
>Yes, correct. That's kind of the point, but you're right that makes it hard to keep in sync.

Agree. Just wanted to be sure.